Makefile

```diff
@@ -85,6 +88,30 @@ update.package: ## Installs the npm requirements for the project
 	@echo 'yarn-path "../oppia_tools/yarn-1.22.15/bin/yarn"' > .yarnrc
 	@echo 'cache-folder "../yarn_cache"' >> .yarnrc
 
+check.container-exists: ## Check if dev-server and angular-build containers exists
+	@run_devserver_prompt="Please, run \`make run-devserver\` (requires internet) once before running \`make run-offline\`"; \
+	containers=$$(docker ps -a --format '{{.Names}}'); \
+	required=(oppia-dev-server oppia-angular-build); \
+	missing=; \
+	for c in $${required[@]}; do \
+		if ! echo "$$containers" | grep -qi $$c; then \
+			missing="$$missing $$c"; \
+		fi; \
+	done; \
+	if [ -n "$$missing" ]; then \
+		echo "Missing containers: $$missing"; \
+		echo $$run_devserver_prompt; \
+		exit 1; \
+	fi
+
+check.dev-server-started-once: ## Check only for dev-server, as if it's started once, applies angular-build is set properly. No need to check explicitely.
+	@COMPLETED_FIRST_RUN=$$(docker ps --all --format="{{json .}}" | grep COMPLETED_FIRST_RUN | jq -r .Labels | sed -n 's/.*COMPLETED_FIRST_RUN=\([^,]*\).*/\1/p'); \
+	if [ "$$COMPLETED_FIRST_RUN" = "0" ]; then \
+		echo "Can't verify if requirements exists in dev-server container."; \
+		echo "Please run \`make dev-server\` (requires internet) once, before running \`make run-offline\`"; \
+		exit 1; \
+	fi
```

Comment:
change this logic to check if the container from the run was healthy or not and updates the label `COMPLETED_FIRST_RUN` based on the result.
also change the cmd to a more appropriate name.